mas-ingestion/mas_ingestion.py

```diff
@@ -0,0 +1,218 @@
+from typing import Dict, Any, List
+import os
+import requests
+import logging
+import json
+
+logging.basicConfig(format='%(asctime)s - %(message)s', level=logging.INFO)
+
+HAS_IDENTIFIER = "$['ods:hasIdentifier']"
+HAS_EVENT = "$['ods:hasEvent']"
+ODS_TYPE = "$['ods:type']"
+SPECIMEN_TYPE = 'https://doi.org/21.T11148/894b1e6cad57e921764e'
+MEDIA_TYPE = 'https://doi.org/21.T11148/bbad8c4e101e8af01115'
+
+
+def build_secret(name: str, secret_key_ref: str) -> Dict[str, str]:
+    return {
+        "schema:name": name,
+        "ods:secretKeyRef": secret_key_ref
+    }
+
+
+def build_attributes(name: str, description: str, image: str, tag: str,
+                     target_filter: Dict[str, Any], batching: bool,
+                     environment: List[Dict[str, Any]],
+                     secrets: List[Dict[str, Any]]) -> \
+        Dict[str, Any]:
+    return {
+        'data': {
+            'type': 'ods:MachineAnnotationService',
+            'attributes': {
+                'schema:name': name,
+                'schema:description': description,
+                'ods:containerImage': image,
+                'ods:containerTag': tag,
+                'ods:TargetDigitalObjectFilter': target_filter,
+                'schema:codeRepository': 'https://github.com/DiSSCo/demo-enrichment-service-image/',
+                'schema:programmingLanguage': 'python',
+                'schema:license': 'https://www.apache.org/licenses/LICENSE-2.0',
+                'ods:batchingPermitted': batching,
+                'ods:hasEnvironmentalVariable': environment,
+                'ods:hasSecretVariable': secrets
+            }
+        }
+    }
+
+
+def bold() -> Dict[str, Any]:
+    name = 'bold-linkage'
+    description = 'Links specimen to an entry in the Barcode of Life Data System (BOLD)'
+    image = 'public.ecr.aws/dissco/bold-linkage'
+    tag = 'latest'
+    target_filter = {
+        HAS_IDENTIFIER: ['*'],
+        ODS_TYPE: [SPECIMEN_TYPE]
+    }
+    batching = False
+    environment = []
+    secrets = [
+        build_secret('API_USER', 'bold-api-user'),
+        build_secret('API_PASSWORD', 'bold-api-password')
```

Comment:
These passwords are not private aren't they?

Reply:
just the name of the secret, not the secret itself :)